app/pages/EditBreakingNewsPage.tsx

```diff
@@ -46,16 +46,25 @@ export const EditBreakingNewsPage = () => {
     ]);
   };
 
-  const onSave = (articleId: string) => {
-    const article = breakingNewsArticles.find(
-      ({ id }: any) => id === articleId
-    );
+  const isActive = (article: NewsArticle): boolean => {
+    const today = new Date();
+    const isEffective = !article.effective_date || new Date(article.effective_date) <= today;
+    const isNotExpired = !article.expiration_date || new Date(article.expiration_date) >= today;
+
+    return Boolean(article.id && isEffective && isNotExpired);
+  };
+
+  const onSave = (index: number) => {
+    const articleFromState = breakingNewsArticles[index];
+    const articleId = articleFromState.id
     const route = `/api/news_articles/${articleId ?? ""}`;
 
-    if (articleId) {
-      dataService.put(route, article);
+    if (articleId){
+      dataService.put(route, articleFromState);
+      alert(`Article "${articleFromState.headline}" has been updated!`)
```

Comment:
These alerts are great! Do you think you could also add one for the delete action? For delete, since it's a dangerous operation, we may want to use [`confirm()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm) instead of `alert()` so that it shows "Cancel" and "Ok" buttons, allowing the user to back out of deleting if they change their mind.